helm/default-apps-aws/templates/apps.yaml

```diff
@@ -6,6 +6,15 @@ kind: App
 metadata:
   annotations:
     chart-operator.giantswarm.io/force-helm-upgrade: "{{ .forceUpgrade }}"
+    {{- if .dependsOn }}
+    {{- if hasPrefix "org-" $.Release.Namespace }}
+    # App is deployed in the org- namespace so the secret name is prefixed by the cluster-id
+    app-operator.giantswarm.io/depends-on: {{ printf "%s-%s" $.Values.clusterID .dependsOn -}}
+    {{- else }}
+    # App is deployed in the cluster-id namespace so prefix is unneeded
+    app-operator.giantswarm.io/depends-on: {{ .dependsOn -}}
+    {{- end }}
+    {{- end }}
```

Comment:
In CAPI installations the namespace is always the organization namespace. There is no need to do this check because the namespace will never be the cluster id.

Reply:
But in my test on golem WC which was named `vpatest` the annotation was working because it was depending on `vpatest-vertical-pod-autoscaler-crd`. It means it was generated thanks to `$.Values.clusterID` right ?
Otherwise we would have got only `vertical-pod-autoscaler-crd` which would not have worked.

Reply:
The else is for vintage

Reply:
Yes, what @QuentinBisson said

Reply:
Oh my bad sorry for the misunderstanding

Reply:
Done.